deeppavlov/utils/benchmarks/benchmarks.py

```diff
@@ -0,0 +1,340 @@
+        if question_idx not in question_indices:
+            output[paragraph_idx]['passage']['questions'].append({
+                'idx': question_idx,
+                'answers': [{'idx': answer_idx, 'label': label}]
+            })
+        else:
+            for question in questions:
+                if question['idx'] == question_idx:
+                    question['answers'].append({'idx': answer_idx, 'label': label})
+
+    submission = list(output.values())
+    return submission
+
+
+def submit_superglue(config: Union[str, Path, dict], output_path: Optional[Union[str, Path]] = None) -> None:
+    """Creates submission file for the SuperGLUE tasks.
+
+    Args:
+        config: Configuration of the model.
+        output_path: Path to output file. If None, file name is selected according corresponding task name.
+
+    """
+
+    config = parse_config(config)
+    data = read_data_by_config(config)
+    iterator = get_iterator_from_config(config, data)
+    task_name = config['dataset_reader']['name']
+
+    data_gen = iterator.gen_batches(1, data_type='test', shuffle=False)
+
+    model = build_model(config)
+
+    submission = []
+
+    if task_name in {'record'}:
+        submission = commonsense_reasoning_prediction(model, data_gen)
+
+    elif task_name in {'copa'}:
+        for idx, (x, _) in enumerate(tqdm(data_gen)):
+            prediction = model.compute(x)[0]
+            label = int(prediction == 'choice2')
+            submission.append({'idx': idx, 'label': label})
+
+    elif task_name in {'multirc'}:
+        submission = multi_sentence_comprehention_prediction(model, data_gen)
+
+    elif task_name in SUPER_GLUE_TASKS:
+        for idx, (x, _) in enumerate(tqdm(data_gen)):
+            prediction = model.compute(x)
+
+            while isinstance(prediction, list):
+                prediction = prediction[0]
+
+            submission.append({'idx': idx, 'label': prediction})
+    else:
+        raise ValueError(f'Unexpected SuperGLUE task name: {task_name}')
+
+    save_path = output_path if output_path is not None else f'{SUPER_GLUE_TASKS[task_name]}.jsonl'
+    save_path = expand_path(save_path)
+    save_path.parent.mkdir(parents=True, exist_ok=True)
+    save_jsonl(submission, save_path)
+    log.info(f'Prediction saved to {save_path}')
+
+
+def submit_rsg(config: Union[str, Path, dict], output_path: Optional[Union[str, Path]] = None) -> None:
+    """Creates submission file for the Russian SuperGLUE tasks.
+
+    Args:
+        config: Configuration of the model.
+        output_path: Path to output file. If None, file name is selected according corresponding task name.
+
+    """
+
+    config = parse_config(config)
+    data = read_data_by_config(config)
+    iterator = get_iterator_from_config(config, data)
+    task_name = config['dataset_reader']['name']
+
+    data_gen = iterator.gen_batches(1, data_type='test', shuffle=False)
+
+    model = build_model(config)
```

Comment:
code is duplicated three times in this file

Reply:
Fixed in [commit](https://github.com/deeppavlov/DeepPavlov/pull/1647/commits/e87eafec9f81c024dcc7820896a3e2502062492d)